labs/b2.md

```diff
@@ -0,0 +1,145 @@
+
+`>` : Redirect stdout to a file.
+
+`>>` : Append stdout to a file.
+
+`<` : Read input from a file.
+
+`|` : Send output from one program to the input of the next.
+
+### Questions
+1. What line could you use to save the first 10 lines of a file that do not contain any vowels to a new file called `return.txt`? 
+2. How could you write `cat output.txt | grep "Cal"` without using a pipe? (Hint: Use a temp file)
+
+### Other useful tricks
+`!!` can be used to repeat the previous command in the shell.
+
+E.g. 
+```
+python program.py = python program.py
+sudo !!           = sudo python program.py
+```
+
+`!:[num]` is treated as the previous command's [num] argument. 
+
+E.g.
+```
+touch doc.txt = touch doc.txt
+vim doc.txt   = vim !:1
+```
+
+Note: This behavior is from bash and some other shells that implement it.
+
+## A quick intro to `vim`
+### Why `vim`?
+ - It's a descendant of vi, which was written in Berkeley by Bill Joy, who went on to found Sun Microsystems.
+ - Sometimes you will be suddenly thrown into `vim` via merging git conflicts or other programs.
+ - It's included in practically every UNIX environment.
+ - You can be very productive when familiar with it.
+
+### Hello World
+     Vim is a modal text editor, meaning that you can change editing modes in order to do different things. There are 3 primarily used modes: **Normal**, **Insert**, and **Visual** mode.
+#### Normal mode:
+ - Used for moving around and editing text
+   - `hjkl` to move left, up, down, and right
+   - `G` to move to end of file, `gg` to move to beginning
+   - `i` to enter **i**nsert mode (`a`, `o` also change mode in different ways)
+   - `dd` to cut a line
+   - `yy` to copy a line
+   - `p` to paste
+   - `/` to search
+   - `u` to undo
+ - Type in commands with `:`
+   - Save with `:w`
+   - Exit with `:q`
+ - Explore more commands online!
+
+#### Insert mode:
+ ![Example](https://d26aqo05ggejx9.cloudfront.net/i.gif)
+ - Used for editing text like a usual editor
+ - Arrow keys to move
+ - Esc to exit to normal mode (lots of people bind it to Caps Lock)
+
+#### Visual mode:
+ - Enter with `v` from normal mode
+ - Used to select text visually
+ - Modify selection with normal mode movement commands
+ - Use `o` to move the cursor to the other side of the selection
+ - Yanking, deleting, and pasting use `y`, `d`, `p` (sound familiar?)
+
+  A key feature of `vim` is **chaining together commands**.
+  Normal mode is essentially a massive amount of shortcuts that you can combine to quickly navigate and edit a file.
+  Want to move down 3 lines?
+  You know that `j` means move down 1 line, so you can use `3j` to move down 3.
+  `d` is for deletion and `w` is to jump to the next word, so what does `dw` do?
+
+### Questions
+ Try playing around with [lab2.md](https://github.com/tanx16/decal-web/blob/master/labs/b2.md) while looking up some new commands. Use `wget` to download it!
```

Comment:
You have to `wget` from `https://raw.githubusercontent.com/tanx16/decal-web/master/labs/b2.md` though, so I'd mention that perhaps.

Reply:
also probably better to link to a more permanent URL (e.g. something with the commit hash in it?) so the link won't break when this changes next semester.